src/client/content_mapblock.cpp

```diff
@@ -1368,11 +1408,20 @@ void MapblockMeshGenerator::drawNodeboxNode()
 
 	// locate possible neighboring nodes to connect to
 	u8 neighbors_set = 0;
-	if (f->node_box.type == NODEBOX_CONNECTED) {
-		for (int dir = 0; dir != 6; dir++) {
-			u8 flag = 1 << dir;
-			v3s16 p2 = blockpos_nodes + p + nodebox_connection_dirs[dir];
-			MapNode n2 = data->m_vmanip.getNodeNoEx(p2);
+	u8 solid_neighbors = 0;
+	u8 sametype_neighbors = 0;
+	for (int dir = 0; dir != 6; dir++) {
+		u8 flag = 1 << dir;
+		v3s16 p2 = blockpos_nodes + p + nodebox_tile_dirs[dir];
+		MapNode n2 = data->m_vmanip.getNodeNoEx(p2);
+		if (n2.param0 == n.param0 && n2.param2 == n.param2)
+			sametype_neighbors |= flag;
```

Comment:
`param2` is only relevant to certain drawtypes and may contain any value when `minetest.swap_node` is used.
I don't think that's an important edge-case and can be omitted if the code would become too complex.
However, I would very much welcome a few comments about the purpose of those variables.

Reply:
I've added detection for rotation and level in param2 + comments